Cp 496 logging package

bogdanbudisan opened the pull request:
added a shared package for writing logs

bogdan-budisan committed:
added a shared package for writing logs

bogdan-budisan committed:
added a shared package for writing logs

bogdan-budisan committed:
removed static from LoggerUtil
added LoggerUtil usage example

bogdan-budisan committed:
removed static from LoggerUtil
added LoggerUtil usage example

bogdan-budisan committed:
changed LoggerUtil usage example

bogdan-budisan committed:
changed LoggerUtil usage example

Commit:
Completed dcumentation

- Modified lints rules
- Added api documentation
- Added Ide configuration

francescoberardi committed:
Merge pull request #5 from francescoberardi/CP-498-documentation

Completed documentation

francescoberardi commented on core/shared/lib/src/logger/abstract_logger.dart:
```diff
@@ -0,0 +1,19 @@
+
+  void error(String message);
+
+  void fatal(String message);
```
We should add documentation comment for  (/// .....)

francescoberardi commented on core/shared/lib/src/logger/logger_util.dart:
```diff
@@ -0,0 +1,78 @@
+// then only this class needs to change, while its usages would stay the same
+class LoggerUtil extends AbstractLogger {
+
+  var logger = Logger();
```
```suggestion
  final Logger logger;
  
    LoggerUtil({required this.logger});
```

We should pass the external dependencies in the constructor

francescoberardi replied:
And then register the external dependency and the LoggerUtil inside the shared_injection_module
```suggestion
    InjectionModule.injector.registerLazySingleton<AbstractLogger>(
        () => LoggerUtil(logger: InjectionModule.injector()));
    InjectionModule.injector.registerLazySingleton<Logger>(() => Logger());
```

francescoberardi commented on features/feature1/lib/src/presentation/pages/feature1_page.dart:
```diff
@@ -28,11 +28,14 @@ class Feature1BodyPage extends StatelessWidget {
     return BlocBuilder<Feature1Bloc, Feature1State>(
       builder: (context, state) {
+        LoggerUtil().info("current state: $state");
+
```
After previous changes here we could access the logger doing something like this:
```suggestion
 InjectionModule.injector<AbstractLogger>().info("current state: $state");

```

francescoberardi commented on core/shared/lib/shared.dart:
```diff
@@ -7,3 +7,4 @@ export './src/module/shared_injection_module.dart';
 export './src/network/network_info.dart';
 export './src/usecases/usecase.dart';
 export './src/usecases/usecase_params.dart';
+export './src/logger/logger_util.dart';
```
We should export only the abstract_logger.dart to the other packages, hiding the real implementation

francescoberardi reviewed (changes requested):
I add some changes proposals.
Let me know if all is clear and If you agree with them.

bogdan-budisan committed:
added documentation comments
moved the LoggerLevel enum to the abstraction side
added Logger to the DI module
changed the exports to share only the abstraction class

bogdan-budisan committed:
added documentation comments
moved the LoggerLevel enum to the abstraction side
added Logger to the DI module
changed the exports to share only the abstraction class

bogdanbudisan commented:
I later decided to move the LoggerLevel enum to the abstract class, to have them as part of the abstraction rather than the implementation, and to avoid having the abstract class depend on the implementation class
If a different library is used, the LoggerLevels should all stay the same as well

But I'm not sure whether the LoggerLevels are expected to change in the future or not (?)

bogdan-budisan committed:
Merge remote-tracking branch 'origin/CP-496-logging-package' into CP-496-logging-package

bogdanbudisan commented:
> 
> 
> I add some changes proposals.
> Let me know if all is clear and If you agree with them.
